tests/hikari/utilities/test_mapping.py

```diff
@@ -0,0 +1,160 @@
+
+    @pytest.mark.asyncio
+    async def test___setitem___garbage_collection(self):
+        mock_map = mapping.MRIMutableMapping(datetime.timedelta(seconds=0.25))
+        mock_map.update({"OK": "no", "blam": "booga"})
+        await asyncio.sleep(0.1)
+        assert mock_map == {"OK": "no", "blam": "booga"}
+        mock_map.update({"ayanami": "rei", "owo": "awoo"})
+        assert mock_map == {"OK": "no", "blam": "booga", "ayanami": "rei", "owo": "awoo"}
+        await asyncio.sleep(0.2)
+        mock_map.update({"nyaa": "qt"})
+        assert mock_map == {"ayanami": "rei", "owo": "awoo", "nyaa": "qt"}
+
+
+class TestCMRIMutableMapping:
+    def test___delitem___for_existing_entry(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        mock_map["Ok"] = 42
+        del mock_map["Ok"]
+        assert "Ok" not in mock_map
+
+    def test___delitem___for_non_existing_entry(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        with pytest.raises(KeyError):
+            del mock_map["Blam"]
+
+    def test___getitem___for_existing_entry(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        mock_map["blat"] = 42
+        assert mock_map["blat"] == 42
+
+    def test___getitem___for_non_existing_entry(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        with pytest.raises(KeyError):
+            mock_map["CIA"]
+
+    def test___iter___(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        mock_map.update({"OK": "blam", "blaaa": "neoeo", "neon": "genesis", "evangelion": None})
+        assert list(mock_map) == ["OK", "blaaa", "neon", "evangelion"]
+
+    def test___len___(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        mock_map.update({"ooga": "blam", "blaaa": "neoeo", "the": "boys", "neon": "genesis", "evangelion": None})
+        assert len(mock_map) == 5
+
+    def test___setitem___when_limit_not_reached(self):
+        mock_map = mapping.CMRIMutableMapping(limit=50)
+        mock_map["OK"] = 523
+        mock_map["blam"] = 512387
+        mock_map.update({"bll": "no", "ieiei": "lslsl"})
+        assert mock_map == {"OK": 523, "blam": 512387, "bll": "no", "ieiei": "lslsl"}
+
+    def test___setitem___when_limit_reached(self):
+        mock_map = mapping.CMRIMutableMapping(limit=4)
+        mock_map.update({"bll": "no", "ieiei": "lslsl", "pacify": "me", "qt": "pie"})
+        mock_map["eva"] = "Rei"
+        mock_map.update({"shinji": "ikari"})
+        assert mock_map == {"pacify": "me", "qt": "pie", "eva": "Rei", "shinji": "ikari"}
+
+
+def test_get_index_or_slice_with_index_within_range():
+    result = mapping.get_index_or_slice({"i": "e", "n": "o", "b": "a", "hikari": "Rei", "p": "a", "o": "o"}, 3)
+    assert result == "Rei"
+
+
+def test_get_index_or_slice_with_index_outside_range():
+    with pytest.raises(IndexError):
+        mapping.get_index_or_slice({"i": "e", "n": "o", "b": "a", "hikari": "noa"}, 77)
```

Comment:
require at least one abridged reference here to get approval.

Reply:
wdym?